### PR TITLE
PIC-836 add court appearances by conviction and CRN

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**
New endpoint to provide court appearances for an event ID, with most recent first.

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/CourtAppearancesResource.java [hunk 1]
```diff
@@ -47,5 +54,25 @@ public CourtAppearanceBasicWrapper getCourtAppearances(@ApiParam(name = "fromDat
...
+            @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "The offender CRN is not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+    @GetMapping(value = "/offenders/crn/{crn}/convictions/{convictionId}/courtAppearances")
+    public CourtAppearanceBasicWrapper getOffenderCourtAppearancesByCrn(final @PathVariable("crn") String crn,
+                                                                        final @PathVariable("convictionId") Long convictionId) {
+
+        List<CourtAppearanceBasic> appearances = offenderService.offenderIdOfCrn(crn)
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/CourtAppearancesResource.java:71]: very minor; could use `final var`

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/CourtAppearancesResource.java (hunk 1 above)
Comment (chris-faulkner-mojdt) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/CourtAppearancesResource.java]: Done. Not that convinced about var when the right hand side doesn't explicitly show the typing. It's probably just the shock of the new :-)

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**